website/content/specification/_index.md

```diff
@@ -1,50 +1,1320 @@
+
+When converted to JSON or YAML, these fields will be collapsed based on their collapse grouping, resulting in the following content instances.
+
+{{< tabs JSON YAML >}}
+{{% tab %}}
+```json {linenos=table,hl_lines=["4-7"]}
+{
+"assembly": {
+  "fields": [
+    {
+      "flag-required": "required 1",
+      "STRVALUE": [ "field-value-1", "field-value-3"]
+    },
+    {
+      "flag-required": "required 2",
+      "STRVALUE": "field-value-2"
+    }
+  ]
+}
+```
+{{% /tab %}}
+{{% tab %}}
+```yaml {linenos=table,hl_lines=["4-7"]}
+---
+assembly:
+  fields:
+  - flag-required: "required 1"
+    STRVALUE:
+    - "field-value-1"
+    - "field-value-3"
+  - flag-required: "required 2"
+    STRVALUE: "field-value-2"
+```
+{{% /tab %}}
+{{% /tabs %}}
+
+If the JSON or YAML instance is converted back to XML, the sequencing of the fields will change due to use of the collapsible feature as follows.
+
+```xml {linenos=table,hl_lines=["2-3"]}
+<assembly>
+  <field flag-required="required 1">field-value-1</field>
+  <field flag-required="required 1">field-value-3</field>
+  <field flag-required="required 2">field-value-2</field>
+</assembly>
+```
+
+In the example above, the fields from the first *collapse group* would appear before the fields from the second *collapse group*. This is because the resulting JSON and YAML instance has no way of indicating the original sequencing, so it must rely on the sequencing provided by the ordering of the *collapse groups* and the values within each group.
+
+### `@default`
+
+The `@default` attribute specifies the default value for the field. When a flag is specified as an optional child of a `<define-field>` or `<define-assembly>`, this value should be considered set for a content instance if the field is omitted in that instance.
+
+A `@default`value MUST only be provided when the data type specified by the `@as-type` is a [simple data types](#simple-data-types). Specifying a `@default` value for a [markup data type](/specification/datatypes/#markup-data-types) MUST result in a Metaschema format error.
+
+Implementations when writing content instances MAY omit writing default values in order to produce a more concise expression of the content.
+
+### `<flag>` Instance Children
+
+A field may have zero or more flag instance children.
+
+See [flag instances](#flag-instances)
+
+### `<define-flag>` Inline Definition
+
+See [inline `<define-flag>`](#inline-define-flag).
+
+### `<json-key>`
+
+TODO: Specify this. Note assembly points to this section.
+
+### JSON Value Keys
+
+TODO: discuss use only when flags may be present.
+
+In XML, the value of a field appears as the text child of the field's element. In JSON and YAML, a property name is needed for the value. The `<json-value-key>` and `<json-value-key-flag>` elements provide a means to control the behavior of how this value is represented.
```

Comment:
```suggestion
In XML, the value of a field appears as the data content (text child or children) of the field's element. In JSON and YAML, a property name is needed for the value. The `<json-value-key>` and `<json-value-key-flag>` elements provide a means to control the behavior of how this value is represented.
```

(In XML, text *children* in cases where content is interrupted by comments or PIs.)

```xml
<title>Twenty thousand <!-- 20000 --> Leagues Under the Sea</title>
```

The value is `Twenty thousand  Leagues Under the Sea` (and note extra space).

Reply:
Indeed. Not sure how that is relevant here. Maybe comments should be discussed in the data types page? Want to suggest something there?

Reply:
Yeah ... certainly not a relevant detail. Just trying to get past the imprecision of "text child" since it is not guaranteed to be so simple.

```suggestion
In XML, the value of a field appears as the textual data content of the field's element. In JSON and YAML, a property name is needed for the value. The `<json-value-key>` and `<json-value-key-flag>` elements provide a means to control the behavior of how this value is represented.
```